components/ai_chat/core/browser/conversation_tools.cc

```diff
@@ -66,19 +66,22 @@ const std::vector<Tool*>& AllTools() {
 
 }  // namespace
 
-ConversationToolProvider::ConversationToolProvider() = default;
+ConversationToolProvider::ConversationToolProvider(
+    base::WeakPtr<Tool> memory_storage_tool)
+    : memory_storage_tool_(memory_storage_tool) {}
 
 ConversationToolProvider::~ConversationToolProvider() = default;
 
 std::vector<base::WeakPtr<Tool>> ConversationToolProvider::GetTools() {
   std::vector<base::WeakPtr<Tool>> tools;
-  if (!features::IsToolsEnabled()) {
-    return tools;
-  }
-
```

Comment:
I only removed it because it's checked too in AllTools, so the behavior is the same before my rebasing on ToolProvider, memory storage tool is still not guarded by this soon to be removed/renamed flag.